app/helpers/concerns/foreman_bootdisk/hosts_helper_ext.rb

```diff
@@ -6,20 +6,33 @@ module ForemanBootdisk::HostsHelperExt
   end
 
   def host_title_actions_with_bootdisk(*args)
+    if @host.bootdisk_downloadable?
+      title_actions(
+        button_group(
+          select_action_button(_('Boot disk'), {:class => 'btn btn-group'},
+            display_bootdisk_link_if_authorized(_("Host '%s' image") % @host.name.split('.')[0], {:controller => 'foreman_bootdisk/disks', :action => 'host', :id => @host}, :class=>'la'),
+            display_bootdisk_link_if_authorized(_("Full host '%s' image") % @host.name.split('.')[0], {:controller => 'foreman_bootdisk/disks', :action => 'full_host', :id => @host}, :class=>'la'),
+            content_tag(:li, "", :class => "divider"),
+            display_bootdisk_link_if_authorized(_("Generic image"), {:controller => 'foreman_bootdisk/disks', :action => 'generic'}, :class=>'la'),
+            display_bootdisk_for_subnet,
+            content_tag(:li, "", :class => "divider"),
+            display_bootdisk_link_if_authorized(_("Help"), {:controller => 'foreman_bootdisk/disks', :action => 'help'}, :class=>'la')
+          )
+        )
+      )
+    else
+      bootdisk_button_disabled
+    end
+
+    host_title_actions_without_bootdisk(*args)
+  end
+
+  def bootdisk_button_disabled
     title_actions(
       button_group(
-        select_action_button(_('Boot disk'), {:class => 'btn btn-group'},
-          display_bootdisk_link_if_authorized(_("Host '%s' image") % @host.name.split('.')[0], {:controller => 'foreman_bootdisk/disks', :action => 'host', :id => @host}, :class=>'la'),
-          display_bootdisk_link_if_authorized(_("Full host '%s' image") % @host.name.split('.')[0], {:controller => 'foreman_bootdisk/disks', :action => 'full_host', :id => @host}, :class=>'la'),
-          content_tag(:li, "", :class => "divider"),
-          display_bootdisk_link_if_authorized(_("Generic image"), {:controller => 'foreman_bootdisk/disks', :action => 'generic'}, :class=>'la'),
-          display_bootdisk_for_subnet,
-          content_tag(:li, "", :class => "divider"),
-          display_bootdisk_link_if_authorized(_("Help"), {:controller => 'foreman_bootdisk/disks', :action => 'help'}, :class=>'la')
-        )
+        link_to(_("Boot disk"), '#', :disabled => true, :class => 'btn btn-default', :title => "Bootdisk download not available for #{@host.architecture.name} architecture")
```

Comment:
1. The test failures likely come from this #name call.
2. Missing string extraction.
3. "Boot disk" is two words.